go-controller/hybrid-overlay/pkg/controller/node_windows.go

```diff
@@ -182,22 +182,54 @@ func (n *NodeController) AddNode(node *kapi.Node) error {
 		return fmt.Errorf("error getting HCN network: %v", err)
 	}
 
-	klog.Infof("Adding a remote subnet route for CIDR '%s' (node: '%s', remote node address: %s, distributed router MAC: %s, VNI: %v).",
-		cidr.String(), node.Name, nodeIP.String(), drMAC.String(), types.HybridOverlayVNI)
-	networkPolicySettings := hcn.RemoteSubnetRoutePolicySetting{
-		// VXLAN virtual network Identifier. Is expected to be 4097 or higher on Windows
-		IsolationId: types.HybridOverlayVNI,
-		// Distributed router/gateway MAC address
-		DistributedRouterMacAddress: drMAC.String(),
-		// Host IP address of the node
-		ProviderAddress: nodeIP.String(),
-		// Prefix used on the destination node
-		DestinationPrefix: cidr.String(),
+	var remoteSubnets []string
+	remoteSubnets = make([]string, 0)
+	for index := range cidrs {
+		klog.Infof("Adding a remote subnet route for CIDR '%s' (node: '%s', remote node address: %s, distributed router MAC: %s, VNI: %v).",
+			cidrs[index].String(), node.Name, nodeIPs[index].String(), drMAC.String(), types.HybridOverlayVNI)
+		networkPolicySettings := hcn.RemoteSubnetRoutePolicySetting{
+			// VXLAN virtual network Identifier. Is expected to be 4097 or higher on Windows
+			IsolationId: types.HybridOverlayVNI,
+			// Distributed router/gateway MAC address
+			DistributedRouterMacAddress: drMAC.String(),
+			// Host IP address of the node
+			ProviderAddress: nodeIPs[index].String(),
+			// Prefix used on the destination node
+			DestinationPrefix: cidrs[index].String(),
+		}
+
+		remoteSubnets = append(remoteSubnets, cidrs[index].String())
+		err := AddRemoteSubnetPolicy(network, &networkPolicySettings)
+		if err != nil {
+			return err
+		}
 	}
+	n.remoteSubnetMap[node.Status.NodeInfo.MachineID] = remoteSubnets
+	return nil
+}
 
-	n.remoteSubnetMap[node.Status.NodeInfo.MachineID] = cidr.String()
+func (n *NodeController) sameIPNets(cidrs []*net.IPNet) bool {
```

Comment:
could use a clearer name... it kind of sounds like it's checking if `cidrs` contains duplicates or something

Reply:
Done